src/code/z_camera.c

```diff
@@ -1,3070 +1,6778 @@
+    Vec3f* at = &camera->at;
+    Vec3f posOffsetTarget;
+    Vec3f atTarget;
+    Vec3f* eye = &camera->eye;
+    PosRot* playerPosRot = &camera->playerPosRot;
+    f32 temp_f2;
+    f32 phi_f16;
+    f32 sp54;
+    f32 phi_f20;
+    f32 temp_f0_4;
+
+    temp_f0_4 = Player_GetHeight(camera->player);
+    posOffsetTarget.x = 0.0f;
+    posOffsetTarget.y = temp_f0_4 + arg2;
+    posOffsetTarget.z = 0.0f;
+
+    if (PREG(76) && arg4) {
+        posOffsetTarget.y -= Camera_CalcSlopeYAdj(&camera->floorNorm, playerPosRot->rot.y, arg1->yaw, OREG(9));
+    }
 
+    if (camera->playerGroundY == camera->playerPosRot.pos.y || camera->player->actor.gravity > -0.1f ||
+        camera->player->stateFlags1 & 0x200000) {
+        *arg3 = Camera_LERPCeilF(playerPosRot->pos.y, *arg3, PCT(OREG(43)), 0.1f);
+        phi_f20 = playerPosRot->pos.y - *arg3;
+        posOffsetTarget.y -= phi_f20;
+        Camera_LERPCeilVec3f(&posOffsetTarget, &camera->posOffset, camera->yOffsetUpdateRate,
+                             camera->xzOffsetUpdateRate, 0.1f);
+    } else {
+        // this condition can basically never be met, due to the decision to call
+        // this function in Camera_Parallel1
+        if (!PREG(75)) {
+            phi_f20 = playerPosRot->pos.y - *arg3;
+            sp54 = OLib_Vec3fDistXZ(at, &camera->eye);
+            phi_f16 = sp54;
+            Math_atan2f(phi_f20, sp54);
+            temp_f2 = Math_tanf(DEG_TO_RAD(camera->fov * 0.4f)) * phi_f16;
+            if (temp_f2 < phi_f20) {
+                *arg3 += phi_f20 - temp_f2;
+                phi_f20 = temp_f2;
+            } else if (phi_f20 < -temp_f2) {
+                *arg3 += phi_f20 + temp_f2;
+                phi_f20 = -temp_f2;
+            }
+            posOffsetTarget.y -= phi_f20;
+        } else {
+            phi_f20 = playerPosRot->pos.y - *arg3;
+            temp_f2 = Math_atan2f(phi_f20, OLib_Vec3fDistXZ(at, eye));
+            if (DEG_TO_RAD(OREG(32)) < temp_f2) {
+                phi_f16 = 1 - sinf(temp_f2 - DEG_TO_RAD(OREG(32)));
+            } else if (temp_f2 < DEG_TO_RAD(OREG(33))) {
+                phi_f16 = 1 - sinf(DEG_TO_RAD(OREG(33)) - temp_f2);
+            } else {
+                phi_f16 = 1;
+            }
+            posOffsetTarget.y -= phi_f20 * phi_f16;
+        }
+        Camera_LERPCeilVec3f(&posOffsetTarget, &camera->posOffset, PCT(OREG(29)), PCT(OREG(30)), 0.1f);
+        camera->yOffsetUpdateRate = PCT(OREG(29));
+        camera->xzOffsetUpdateRate = PCT(OREG(30));
+    }
+    atTarget.x = playerPosRot->pos.x + camera->posOffset.x;
+    atTarget.y = playerPosRot->pos.y + camera->posOffset.y;
+    atTarget.z = playerPosRot->pos.z + camera->posOffset.z;
+    Camera_LERPCeilVec3f(&atTarget, at, camera->atLERPStepScale, camera->atLERPStepScale, 0.2f);
     return 1;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_80045C74.s")
+#define FLG_OFFGROUND (1 << 7)
+/**
+ * Adjusts at position for Camera_Battle1 and Camera_KeepOn1
```

Comment:
are there more of these flags? might make more sense to define them grouped together at the top of the file with the other defines

Reply:
I will be adding more as I learn about them, this was the only one I was 100% sure on at the time though.

Reply:
cool, yea makes sense. grouping these with the other defines is probably a good idea, idk if there were any more throughout the file